### PR TITLE
Count Remaining Characters Widget - Test Page

### PR DESCRIPTION
**Pull request by dtaylor113**
image

### CODE REVIEW

File: dist/css/patternfly.css [hunk 1]
```diff
@@ -9035,6 +9035,14 @@ fieldset[disabled] .form-control {
 label {
   font-weight: 600;
 }
+.chars-remaining-pf span {
+  vertical-align: baseline;
```
Comment (rhamilto) [inline dist/css/patternfly.css]: I don't think this is needed as the default value is already baseline.  https://developer.mozilla.org/en-US/docs/Web/CSS/vertical-align

File: dist/css/patternfly.css [hunk 2]
```diff
@@ -9035,6 +9035,14 @@ fieldset[disabled] .form-control {
...
+  padding-right: 5px;
+  font-weight: 600;
+}
+.chars-warn-remaining-pf {
+  color: red;
```
Comment (rhamilto) [inline dist/css/patternfly.css]: This should use a variable.  Most likely @brand-danger.

File: less/forms.less [hunk 3]
```diff
@@ -31,3 +31,13 @@
 label {
   font-weight: 600;
 }
+
+.chars-remaining-pf span {
+  vertical-align: baseline;
```
Comment (rhamilto) [inline less/forms.less]: I don't think this is needed as the default value is already baseline. https://developer.mozilla.org/en-US/docs/Web/CSS/vertical-align

File: less/forms.less [hunk 4]
```diff
@@ -31,3 +31,13 @@
...
+  padding-right: 5px;
+  font-weight: 600;
+}
+
+.chars-warn-remaining-pf {
+  color: red;
```
Comment (rhamilto) [inline less/forms.less]: This should use a variable. Most likely @brand-danger.

File: less/forms.less [hunk 5]
```diff
@@ -31,3 +31,13 @@
 label {
   font-weight: 600;
 }
+
+.chars-remaining-pf span {
```
Comment (rhamilto) [inline less/forms.less]: These rules should go first in the file (alphabetize rules by selector).

File: tests-src/forms.html [hunk 6]
```diff
@@ -24,6 +24,38 @@
           </div>
         </div>
       </form>
+      <h2>Count Remaining Characters Control</h2>
+      <form>
+        <div class="form-group">
+          <label for="messageArea"></label>
+          <textarea class="form-control" id="messageArea" name="text" placeholder="Type in your message" rows="5"></textarea>
```
Comment (rhamilto) [inline tests-src/forms.html:31]: Please change all camel case instances to follow http://codeguide.patternfly.org/#css-classes

File: less/forms.less [hunk 7]
```diff
@@ -31,3 +31,13 @@
...
+  vertical-align: baseline;
+  padding-right: 5px;
```
Comment (rhamilto) [inline less/forms.less]: Please alphabetize per http://codeguide.patternfly.org/#css-declaration-order

File: tests-src/forms.html (hunk 6 above)
Comment (dtaylor113) [inline tests-src/forms.html:31]: Hi, while I agree that class names should use dashes, I don't think ids or variables should use them; they should be camel case.  I followed the already existing precedence shown in dist/patternfly.js for ids and variable names.

File: less/forms.less (hunk 3 above)
Comment (dtaylor113) [inline less/forms.less]: fixed

File: less/forms.less (hunk 4 above)
Comment (dtaylor113) [inline less/forms.less]: fixed

File: less/forms.less (hunk 5 above)
Comment (dtaylor113) [inline less/forms.less]: fixed

File: less/forms.less (hunk 7 above)
Comment (dtaylor113) [inline less/forms.less]: fixed

File: tests-src/forms.html (hunk 6 above)
Comment (dtaylor113) [inline tests-src/forms.html:31]: Note that the PatternFly Examples section directly above this new section uses camel case for IDs.

File: less/forms.less [hunk 8]
```diff
@@ -1,6 +1,14 @@
 //
 // Forms
 // --------------------------------------------------
+.chars-remaining-pf span {
+  padding-right: 5px;
+  font-weight: 600;
```
Comment (rhamilto) [inline less/forms.less:5]: Alpha fix, please.  (font-weight should come before padding-right per http://codeguide.patternfly.org/#css-declaration-order)

Comment (dtaylor113) [inline less/forms.less:5]: fixed

**Pull request merged**